src/obr/cli.py

```diff
@@ -292,6 +308,148 @@ def query(ctx: click.Context, **kwargs):
+    required=False,
+    is_flag=True,
+    help=(
+        "If set, .log files will not be archived. This does not affect .log files"
+        " passed via the --file option."
+    ),
+)
+@click.option(
+    "-a",
+    "--file",
+    required=False,
+    multiple=True,
+    help="Path(s) to non-logfile(s) to be also added to the repository.",
+)
+@click.pass_context
+def archive(ctx: click.Context, **kwargs):
+    if current_path := kwargs.get("folder", "."):
+        os.chdir(current_path)
+        current_path = Path(current_path).absolute()
+
+    # setup project and jobs
+    project = OpenFOAMProject().init_project()
+    filters: tuple[str] = kwargs.get("filter", ())
+    jobs = filter_jobs_by_query(project, filters)
+
+    time = str(datetime.now()).replace(" ", "_")
+    target_folder: str = kwargs.get("repo", "")
+    use_github_repo = False
+    repo = None
+    branch_name = None
+    previous_branch = None
+    # check if given path is actually a github repository
+    try:
+        repo = Repo(path=target_folder)
+        previous_branch = repo.active_branch.name
+        branch_name = "archive-" + (
+            str(datetime.now()).rsplit(":", 1)[0].replace(" ", ":").replace(":", "_")
+        )
+        use_github_repo = True
+        logging.info(f"checkout archive-{branch_name}")
+        repo.git.checkout("HEAD", b=branch_name)
+    except InvalidGitRepositoryError:
+        logging.warn(
+            f"Given directory {target_folder=} is not a github repository. Will only"
+            " copy files."
+        )
+
+    # setup target folder
+    path = Path(target_folder + f"/{time}")
+    if not path.exists():
+        logging.info(f"creating {path}")
+        path.mkdir()
+
+    skip = kwargs.get("skip_logs")
+    if not skip:
+        # iterate cases and copy log files into target repo
+        for job in jobs:
+            case_folder = Path(job.path) / "case"
+            if not case_folder.exists():
+                logging.info(f"Job with {job.id=} has no case folder.")
+                continue
+
+            # skip if either the most recent obr action failed or the label is set to "not success"
+            case = OpenFOAMCase(str(case_folder), job)
+            if not case.was_successful():
+                logging.info(
+                    f"Skipping Job with {job.id=} due to recent failure states."
+                )
+                continue
+
+            root, _, files = next(os.walk(case_folder))
+            for file in files:
+                log_file = Path(root) / file
+                if log_file.is_relative_to(current_path):
+                    log_file = log_file.relative_to(current_path)
+                if file.endswith("log"):
+                    target_file = path
+                    if target_file.is_relative_to(current_path):
+                        target_file = target_file.relative_to(current_path)
+                    copy_to_archive(repo, use_github_repo, log_file, target_file)
```

Comment:
See my comment above, the idea is to copy workspace/<job.id>/case/<files> to repo/folder/<workspace>/job.id/case/<files> and keep the workspace structure intact.